src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceEditorBackgroundLinkHighlighter.java

```diff
@@ -351,8 +354,40 @@ private void navigateToUrl(String url)
       if (reSrcRef.test(url))
          return;
 
-      // treat other URLs as paths to files on the server
       final String finalUrl = url;
+      
+      // github issues for this package (uses BugReports)
+      Pattern reIssue = Pattern.create("^#[0-9]+$");
+      if (reIssue.test(url))
+      {
+         server_.getIssueUrl(url, new SimpleRequestCallback<String>()
+         {
+            @Override
+            public void onResponseReceived(String response)
+            {
+               if (response.length() > 0)
+                  globalDisplay_.openWindow(response); 
+               else 
+                  RStudioGinjector.INSTANCE.getGlobalDisplay().showErrorMessage(
+                     "Could not resolve " + finalUrl + ". Please make sure this is an R package project with a BugReports field set.");
+            }
+         });
+
+         return;
+      }
+
+      // explicit github issues
+      Pattern reSpecificGithubIssue = Pattern.create("[-a-zA-Z0-9]+/[-a-zA-Z0-9]+#[0-9]+");
+      if (reSpecificGithubIssue.test(url))
+      {
+         String[] parts = url.split("#");
+         String issueUrl = "https://www.github.com/" + parts[0] + "/issues/" + parts[1];
```

Comment:
Is there a way for us to leave the door open for projects hosted in other locations (e.g. GitLab or whatever else)?

Reply:
Maybe `remotes` can help: https://remotes.r-lib.org/articles/dependencies.html ?

Reply:
But most of the time, we just need to link within the self repo. Linking to another repo's issue is pretty rare (cc @DavisVaughan), and we might as well just support plain url there, i.e. `test_that("this thing works (https://www.example.com)", {`

these currently are not allowed because the url has to be outside a string, but perhaps we can special case here.

Reply:
Right, but my point here is (theoretically) someone might use issue markers for issues that are hosted in e.g. GitLab rather than GitHub, and so we would want to make sure to support issue trackers other than GitHub. Presumedly those would use the same syntax for identifying issues.

Reply:
> Should we be using the BugReports URL here? 

I don't think so, because the format `org/repo#123` implies github, perhaps we can support something like `gitlab::org/repo#123` following what `remotes` does.